### PR TITLE
V1.2.0 rc

### PR DESCRIPTION
**Pull request by ecurley-nextcentury**
This includes all changes required for the M36 dry run and evaluation. Once this is in master, I will create the following artifacts:
* Packaged java release on OSSRH
* Validation docker image on github
* Github release
* (possibly) Packaged python release on pypi

### CODE REVIEW

**Comment by ecurley-nextcentury** [comment]
@gitnordt, I had to bump the version to 1.2.1 due to that erroneous submission I made to OSSRH a couple weeks ago

File: docker/aif-validator/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+### Running the container
+
+To run the AIF Validator Docker container, the `run.sh.example` script shows one way to run it.  You can copy the `run.sh.example` script to a new file, `run.sh`. The run script will start the Docker container. Before executing the script, update the passed in Docker environment variables within the `run.sh` script. These environment variables should be configured to meet your needs for your particular AIF Validator execution. Each variable is described in the table below.
+
+
+| Env Variable            | Description                                                    | Default              |
+| :---------------------- | :------------------------------------------------------------- | :------------------- |
+| `VALIDATION_HOME`       | Location in the container for the built, git clone of AIF      | `/opt/aif-validator` |
+| `VALIDATION_FLAGS`      | Flags sent to the AIF validator (see below)                    | none                 |
+| `VALIDATION_LOG`        | `stdout` or a file in the mounted directory to save the log    | `stdout`             |
```
Comment (gitnordt) [inline docker/aif-validator/README.md:54]: The file has '/v/validate_log.out' as the default. Please update here or in the run.sh.example.

File: docker/aif-validator/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+| `VALIDATE_DIR_OR_FILES` | Either `directory` or `files`                                  | `directory`          |
+| `TARGET_TO_VALIDATE`    | The directory to validate OR the space separated list of files | `${pwd}`             |
```
Comment (gitnordt) [inline docker/aif-validator/README.md:56]: The file has '/v' as the default. Please update here or in the run.sh.example.

File: docker/aif-validator/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,61 @@
+FROM amazonlinux:latest
+MAINTAINER craig.warsaw@caci.com
```
Comment (gitnordt) [inline docker/aif-validator/Dockerfile:2]: Should Craig's email address be changed to yours? I also saw Craig referenced in the AIDA-Interchange-Format/README.md.

File: python/tests/scaling_test.py [hunk 4]
```diff
@@ -183,62 +181,183 @@ def write_to_file(self, testname):
...
+        "Entity",
+        "Place",
+        "Beneficiary",
+        "Giver",
+        "Recipient",
+        "Thing",
+        "Time"]
 
 if __name__ == "__main__":
     ScalingTest().run_scaling_test()
```
[the remark this answers is not in the document]
Comment (ecurley-nextcentury) [reply python/tests/scaling_test.py]: Fixed

File: java/src/main/resources/com/ncc/aif/restricted_aif.shacl [hunk 5]
```diff
@@ -241,3 +230,106 @@ aida:LinkAssertionShape
...
+            $this aida:prototype / rdf:type ?prototypeType .
+            FILTER ( ?memberType != ?prototypeType )
+        }
+    """
+    .
+
+aida:ClusterMembersShape
+    a sh:NodeShape ;
+    sh:targetClass aida:SameAsCluster;
+    sh:sparql aida:ClusterMembersSameAsBaseClass .
```
[the remark this answers is not in the document]
Comment (ecurley-nextcentury) [reply java/src/main/resources/com/ncc/aif/restricted_aif.shacl]: Fixed

File: java/src/main/java/com/ncc/aif/util/AIFComparableNode.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+             return rep.equals(other.rep);
+         } else if (node != null) {
+             return node == other.node;
+         } else {
+             // everything is null, thus equal
+             return true;
+         }
+     }
+ }
+
```
[the remark this answers is not in the document]
Comment (ecurley-nextcentury) [reply java/src/main/java/com/ncc/aif/util/AIFComparableNode.java]: Fixed

File: java/src/main/java/com/ncc/aif/InstrumentedValidationEngine.java [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+		Dataset dataset = ARQFactory.get().getDataset(dataModel);
+		dataset.addNamedModel(shapesGraphURI.toString(), shapesModel);
+
+		ShapesGraph shapesGraph = new ShapesGraph(shapesModel);
+
+		ValidationEngine engine = new InstrumentedValidationEngine(dataset, shapesGraphURI, shapesGraph, null);
+		engine.setConfiguration(configuration);
+		return engine;
+	}
+}
```
[the remark this answers is not in the document]
Comment (ecurley-nextcentury) [reply java/src/main/java/com/ncc/aif/InstrumentedValidationEngine.java]: Fixed

File: python/README.md [hunk 8]
```diff
@@ -1,22 +1,80 @@
...
+
+## Create virtual environment
+
+When you are starting development of a python project, you first need to create a virtual environment. The name of the virtual environment in the example below, `aida-interchange-format`, assumes you are making changes or testing the AIF library. Feel free to use a name specific to your application if you are just using the AIF library.
+
+To create the virtual environment and install the latest AIF, run the following:
+
+```bash
+$ virtualenv -p python3 ~/.virtualenvs/aida-interchange-format
+$ source ~/.virtualenvs/aida-interchange-format/bin/activate
```
[the remark this answers is not in the document]
Comment (ecurley-nextcentury) [reply python/README.md:60]: Good point. Removed.

File: python/README.md [hunk 9]
```diff
@@ -1,22 +1,80 @@
...
+$ pip install aida-interchange
```
[the remark this answers is not in the document]
Comment (ecurley-nextcentury) [reply python/README.md:61]: No. I'm assuming they will want the latest version. Leaving this.

File: python/README.md [hunk 10]
```diff
@@ -1,22 +1,80 @@
...
 
-## Install setup.py
+## Install
 
-To install `aida_interchange`, make sure a [Python Virtual Environment](VIRTUAL-README.md) is activated, navigate to the `AIDA-Interchange-Format/python` directory, and run the following command:
+To install `aida_interchange`, make sure a [Python Virtual Environment](#virtualenv) is activated and run the following command:
 
 ```bash
-$ python3 setup.py install
+$ pip install aida-interchange==x.x.x
```
[the remark this answers is not in the document]
Comment (ecurley-nextcentury) [reply python/README.md]: I've removed the versions altogether.

File: docker/aif-validator/README.md [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+
+### Running the container
+
+To run the AIF Validator Docker container, the `run.sh.example` script shows one way to run it.  You can copy the `run.sh.example` script to a new file, `run.sh`. The run script will start the Docker container. Before executing the script, update the passed in Docker environment variables within the `run.sh` script. These environment variables should be configured to meet your needs for your particular AIF Validator execution. Each variable is described in the table below.
+
+
+| Env Variable            | Description                                                    | Default              |
+| :---------------------- | :------------------------------------------------------------- | :------------------- |
+| `VALIDATION_HOME`       | Location in the container for the built, git clone of AIF      | `/opt/aif-validator` |
+| `VALIDATION_FLAGS`      | Flags sent to the AIF validator (see below)                    | none                 |
```
[the remark this answers is not in the document]
Comment (ecurley-nextcentury) [reply docker/aif-validator/README.md:53]: The defaults here are the defaults for the image, not for the script. Leaving.

File: docker/aif-validator/README.md (hunk 1 above)
Comment (ecurley-nextcentury) [reply docker/aif-validator/README.md:54]: The defaults here are the defaults for the image, not for the script. Leaving.

File: docker/aif-validator/Dockerfile (hunk 3 above)
Comment (ecurley-nextcentury) [reply docker/aif-validator/Dockerfile]: Good catch. I've removed MAINTAINER as it's deprecated and I've changed README.md to use aif-support@nextcentury.com

**Comment by ecurley-nextcentury** [comment]
@gitnordt, I think I've addressed all your concerns, but I've also made some changes extraneous to your suggestions. They should probably be vetted as well. Let me know if you have any questions

File: docker/aif-validator/README.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (gitnordt) [reply docker/aif-validator/README.md:53]: Ok, just wanted to make sure you were aware.

**Pull request merged**